packages/atomic/src/directives/hierarchical-path.spec.ts

```diff
@@ -0,0 +1,216 @@
+import {html, render} from 'lit';
+import {describe, beforeEach, afterEach, it, expect} from 'vitest';
+import {
+  hierarchicalPath,
+  type HierarchicalPathProps,
+} from './hierarchical-path';
+
+describe('hierarchicalPath', () => {
+  let container: HTMLElement;
+
+  beforeEach(() => {
+    container = document.createElement('div');
+    document.body.appendChild(container);
+  });
+
+  afterEach(() => {
+    document.body.removeChild(container);
+  });
+
+  const renderPath = (props: HierarchicalPathProps) => {
+    render(html`<div>${hierarchicalPath(props)}</div>`, container);
+  };
+
+  it('should render empty content for empty path', () => {
+    renderPath({path: []});
+
+    expect(container.textContent?.trim()).toBe('');
+  });
+
+  it('should render custom empty path content', () => {
+    renderPath({
+      path: [],
+      emptyPathContent: html`<span>No path</span>`,
+    });
+
+    expect(container.textContent).toContain('No path');
+  });
+
+  it('should render single path item without separator', () => {
+    renderPath({path: ['electronics']});
+
+    expect(container.textContent).toContain('electronics');
+    expect(container.querySelectorAll('span').length).toBe(1);
+  });
+
+  it('should render multiple path items with separators', () => {
+    renderPath({path: ['electronics', 'computers', 'laptops']});
+
+    expect(container.textContent).toContain('electronics');
+    expect(container.textContent).toContain('computers');
+    expect(container.textContent).toContain('laptops');
+    expect(container.textContent).toContain('/');
+  });
+
+  it('should use custom separator', () => {
+    renderPath({
+      path: ['electronics', 'computers'],
+      separator: ' > ',
+    });
+
+    expect(container.textContent?.replace(/\s+/g, ' ').trim()).toContain(
+      'electronics > computers'
+    );
+  });
+
+  it('should apply default CSS classes', () => {
+    renderPath({path: ['electronics', 'computers']});
+
+    const spans = container.querySelectorAll('span');
+    const separatorSpan = spans[1]; // First separator
+    const itemSpan = spans[0]; // First item
+
+    expect(separatorSpan).toHaveClass('mx-0.5');
+    expect(itemSpan).toHaveClass('max-w-max');
+    expect(itemSpan).toHaveClass('flex-1');
+    expect(itemSpan).toHaveClass('truncate');
+  });
+
+  it('should apply custom CSS classes', () => {
+    renderPath({
+      path: ['electronics', 'computers'],
+      separatorClass: 'custom-separator',
+      itemClass: 'custom-item',
+    });
+
+    const spans = container.querySelectorAll('span');
+    const separatorSpan = spans[1];
+    const itemSpan = spans[0];
+
+    expect(separatorSpan).toHaveClass('custom-separator');
+    expect(itemSpan).toHaveClass('custom-item');
+  });
+
+  it('should not ellipse short paths', () => {
+    renderPath({
+      path: ['electronics', 'computers', 'laptops'],
+      maxLength: 3,
+    });
+
+    expect(container.textContent).not.toContain('...');
```

Comment:
Similar to comments I made previously, we could use the `expect(container.textContent?.replace(/\s+/g, " ").trim()).toBe(/* ... */));` approach in the ellipsis tests.